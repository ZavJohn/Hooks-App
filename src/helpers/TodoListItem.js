import React from 'react'

export const TodoListItem = ({ todo, handleDelete, handleToggle, index }) => {
  return (
    <li
        key = { todo.id }
        className="list-group-item"
    > 
        <p className={`text-start ${ todo.done && 'complete'}`}
            onClick={ () => handleToggle ( todo.id )}
        >
                { index + 1 }.{ todo.desc }
            </p>
        <div className="d-grid gap-2 d-md-flex justify-content-md-end">
            <button
                className="btn btn-danger btn-sm"
                onClick={ () => handleDelete( todo.id ) }
            >
                Borrar
            </button>
        </div>
    </li>
  )
}
