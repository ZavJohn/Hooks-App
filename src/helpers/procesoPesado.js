export const procesoPesado = ( iteraciones ) => {

    for ( let i = 0; i < iteraciones; i ++){

        console.log('Inicia el proceso pesado');
    }

    return `${ iteraciones } iteraciones realizadas`;
}