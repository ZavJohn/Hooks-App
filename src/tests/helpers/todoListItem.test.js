
import React from 'react';
import { shallow } from 'enzyme';

import { TodoListItem } from '../../helpers/TodoListItem';
import { demosTodos } from '../fixtures/demosTodos';
import '@testing-library/jest-dom';

describe('Pruebas del <todoListItem/>', () => {

    const handleDelete = jest.fn();
    const handleToggle = jest.fn();
    
    const wrapper = shallow(
        <TodoListItem 
            todo= { demosTodos[0] }
            handleDelete= { handleDelete }
            handleToggle= { handleToggle }
            index={0}
        />);

    test('Debe mostrarse correctamente', () => { 

        expect( wrapper ).toMatchSnapshot();
     });

     test('Debe llamar el handdleDelete', () => { 

        wrapper.find('button').simulate('click');
        expect( handleDelete ).toHaveBeenCalledWith( demosTodos[0].id );
        
     });

     test('Debe llamar el handleToggle', () => { 

        expect( wrapper ).toMatchSnapshot();wrapper.find('p').simulate('click');
        expect( handleToggle ).toHaveBeenCalledWith( demosTodos[0].id );

     });

     test('Debe mostrar el texto correctamente', () => { 

        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe(`1.${ demosTodos[0].desc }`);
     });

     test('Debe tener la clase complete', () => { 

        const todo = demosTodos[0];
        todo.done = true;

        const wrapper = shallow(
            <TodoListItem 
                todo= { todo }
            />);
        expect( wrapper.find('p').hasClass('complete') ).toBe(true);
     });



});