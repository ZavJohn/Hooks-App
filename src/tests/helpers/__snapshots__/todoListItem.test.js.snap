// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del <todoListItem/> Debe llamar el handleToggle 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="text-start false"
    onClick={[Function]}
  >
    1
    .
    Aprender React
  </p>
  <div
    className="d-grid gap-2 d-md-flex justify-content-md-end"
  >
    <button
      className="btn btn-danger btn-sm"
      onClick={[Function]}
    >
      Borrar
    </button>
  </div>
</li>
`;

exports[`Pruebas del <todoListItem/> Debe mostrarse correctamente 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="text-start false"
    onClick={[Function]}
  >
    1
    .
    Aprender React
  </p>
  <div
    className="d-grid gap-2 d-md-flex justify-content-md-end"
  >
    <button
      className="btn btn-danger btn-sm"
      onClick={[Function]}
    >
      Borrar
    </button>
  </div>
</li>
`;
